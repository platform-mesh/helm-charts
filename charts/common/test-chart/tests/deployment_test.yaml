suite: deployment
templates:
  - deployment.yaml
values:
  - ../values.yaml
chart:
  namespace: default
  appVersion: 0.1.0
tests:
  - it: test tag default
    asserts:
      - matchSnapshot: {}
  - it: test tag override
    set:
      image:
        tag: 1.0.0
    asserts:
      - matchSnapshot: {}
  - it: test name override
    set:
      image:
        name: my-image
    asserts:
      - matchSnapshot: {}
  - it: test sentry enabled
    set:
      image:
        name: my-image
      global:
        sentry:
          enabled: true
      environment: "test"
      sentry:
        environment: "test"
    asserts:
      - matchSnapshot: {}
  - it: test extras
    set:
      extraEnvs:
        - name: MY_ENV
          value: my-value
      extraArgs:
        - --arg1=value1
        - --arg2=value2
      extraVolumes:
        - name: my-volume
          emptyDir: {}
        - name: my-volume2
          emptyDir: {}
      extraVolumeMounts:
        - name: my-volume
          mountPath: /mnt/my-volume
        - name: my-volume2
          mountPath: /mnt/my-volume2
    asserts:
      - matchSnapshot: {}
  - it: test global region
    set:
      global:
        region: us-east-1
    asserts:
      - matchSnapshot: {}
  - it: set cpu limits
    set:
      deployment:
        resources:
          limits:
            cpu: 1000m
    asserts:
      - matchSnapshot: {}
  - it: test tracing
    set:
      tracing:
        enabled: true
        collector:
            endpoint: collector:4317
    asserts:
      - matchSnapshot: {}
  - it: test duplicate ports
    set:
      health:
        port: 8080
    asserts:
      - matchSnapshot: {}
