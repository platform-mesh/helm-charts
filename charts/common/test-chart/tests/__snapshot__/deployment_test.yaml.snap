set cpu limits:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-test-chart
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          labels:
            app: RELEASE-NAME-test-chart
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - -- operator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=0.1.0
                - --image-name="ghcr.io/your-org/your-image"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env: null
              image: ghcr.io/your-org/your-image:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-test-chart
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 1000m
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-test-chart
          terminationGracePeriodSeconds: 10
          volumeMounts: null
          volumes: null
test extras:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-test-chart
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          labels:
            app: RELEASE-NAME-test-chart
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - -- operator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=0.1.0
                - --image-name="ghcr.io/your-org/your-image"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
                - --arg1=value1
                - --arg2=value2
              env:
                - name: MY_ENV
                  value: my-value
              image: ghcr.io/your-org/your-image:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-test-chart
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-test-chart
          terminationGracePeriodSeconds: 10
          volumeMounts:
            - mountPath: /mnt/my-volume
              name: my-volume
            - mountPath: /mnt/my-volume2
              name: my-volume2
          volumes:
            - emptyDir: {}
              name: my-volume
            - emptyDir: {}
              name: my-volume2
test global region:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-test-chart
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          labels:
            app: RELEASE-NAME-test-chart
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - -- operator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=us-east-1
                - --environment=local
                - --image-tag=0.1.0
                - --image-name="ghcr.io/your-org/your-image"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env: null
              image: ghcr.io/your-org/your-image:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-test-chart
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-test-chart
          terminationGracePeriodSeconds: 10
          volumeMounts: null
          volumes: null
test name override:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-test-chart
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          labels:
            app: RELEASE-NAME-test-chart
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - -- operator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=0.1.0
                - --image-name="my-image"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env: null
              image: my-image:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-test-chart
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-test-chart
          terminationGracePeriodSeconds: 10
          volumeMounts: null
          volumes: null
test sentry enabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-test-chart
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          labels:
            app: RELEASE-NAME-test-chart
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - -- operator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=test
                - --image-tag=0.1.0
                - --image-name="my-image"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env:
                - name: SENTRY_DSN
                  valueFrom:
                    secretKeyRef:
                      key: dsn
                      name: RELEASE-NAME-test-chart-sentry
              image: my-image:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-test-chart
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-test-chart
          terminationGracePeriodSeconds: 10
          volumeMounts: null
          volumes: null
test tag default:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-test-chart
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          labels:
            app: RELEASE-NAME-test-chart
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - -- operator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=0.1.0
                - --image-name="ghcr.io/your-org/your-image"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env: null
              image: ghcr.io/your-org/your-image:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-test-chart
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-test-chart
          terminationGracePeriodSeconds: 10
          volumeMounts: null
          volumes: null
test tag override:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-test-chart
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          labels:
            app: RELEASE-NAME-test-chart
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - -- operator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=1.0.0
                - --image-name="ghcr.io/your-org/your-image"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env: null
              image: ghcr.io/your-org/your-image:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-test-chart
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-test-chart
          terminationGracePeriodSeconds: 10
          volumeMounts: null
          volumes: null
test tracing:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-test-chart
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          labels:
            app: RELEASE-NAME-test-chart
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - -- operator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=warn
                - --region=local
                - --environment=local
                - --image-tag=0.1.0
                - --image-name="ghcr.io/your-org/your-image"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
                - --tracing-enabled=true
                - --tracing-config-service-name=RELEASE-NAME-test-chart.NAMESPACE
                - --tracing-config-service-version=0.1.0
                - --tracing-config-collector-endpoint=collector:4317
              env: null
              image: ghcr.io/your-org/your-image:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: RELEASE-NAME-test-chart
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-test-chart
          terminationGracePeriodSeconds: 10
          volumeMounts: null
          volumes: null
